src/exchange.rs

```diff
@@ -18,22 +53,37 @@ impl<'a, T: FormattableCurrency> Exchange<'a, T> {
 
     /// Update an ExchangeRate or add it if does not exist.
     pub fn set_rate(&mut self, rate: &'a ExchangeRate<T>) {
-        let key = Exchange::generate_key(rate.from, rate.to);
-        self.map.insert(key, *rate);
+        let query = ExchangeRateQuery::new(rate.from, rate.to);
```

Comment:
the default behavior of always setting the inverse rate might not be desirable. for instance, if there is an exchange rate spread then the inverse rate is not always the same as the forward rate. 

do you think there's a reasonable interface where we can make setting the exact inverse an optional parameter?

Reply:
That sounds reasonable to me. I'm envisioning two options for this:
```
pub fn set_rate(&mut self, rate: &'a ExchangeRate<T>, set_inverse: bool)
```
which forces the user to explicitly pass in whether or not they want the inverse which feels a bit un-ergonomic to me and the other option is to add a new function with the following signature:
```
pub fn set_rate_and_inverse_rate(&mut self, rate: &'a ExchangeRate<T>)
```
which would both set the rate as well as set the inverse rate.

Reply:
let's go with the latter, it also avoids a breaking api change